src/OrbitGl/ThreadTrack.cpp

```diff
@@ -468,19 +468,24 @@ void ThreadTrack::UpdatePrimitives(Batcher* batcher, uint64_t min_tick, uint64_t
                   app_->GetFunctionIdToHighlight(), app_->GetGroupIdToHighlight());
 
   absl::MutexLock lock(&scope_tree_mutex_);
-
   for (const auto& [depth, ordered_nodes] : scope_tree_.GetOrderedNodesByDepth()) {
-    auto first_node_to_draw = ordered_nodes.lower_bound(min_tick);
-    if (first_node_to_draw != ordered_nodes.begin()) --first_node_to_draw;
-
+    if (ordered_nodes.empty()) {
+      continue;
+    }
     track_data_->UpdateMaxDepth(depth);
-
     float world_timer_y = GetYFromDepth(depth - 1);
+
     uint64_t next_pixel_start_time_ns = min_tick;
 
-    for (auto it = first_node_to_draw; it != ordered_nodes.end() && it->first < max_tick; ++it) {
-      const orbit_client_protos::TimerInfo& timer_info = *it->second->GetScope();
-      if (timer_info.end() <= next_pixel_start_time_ns) continue;
+    auto node_to_draw = ordered_nodes.upper_bound(min_tick);
+    // The previous node should be drawn only if its ending is after min_tick.
+    if (node_to_draw != ordered_nodes.begin()) --node_to_draw;
+    if (node_to_draw->second->GetScope()->end() < min_tick) {
```

Comment:
Nit: I prefer to move the comment to be just above this line.

Reply:
The comment was rephrased